en/05_Contributing/04_Release_Process.md

```diff
@@ -154,55 +154,75 @@ How to deprecate an API:
 
 * Add a `@deprecated` item to the docblock tag, with a `{@link <class>}` item pointing to the new API to use.
 * Update the deprecated code to throw a [Deprecation::notice()](api:SilverStripe\Dev\Deprecation::notice()) error.
-* Both the docblock and error message should contain the **target version** where the functionality is removed.
-  So, if you're committing the change to a *3.1* minor release, the target version will be *4.0*. 
-* Deprecations should not be committed to patch releases
+* Both the docblock and error message should contain the **version** where the functionality is deprecated from.
+  So, if you're committing the change to a *4.12* minor release, the target version will be *4.12.0*. 
 * Deprecations should only be committed to pre-release branches, ideally before they enter the "beta" phase.
   If deprecations are introduced after this point, their target version needs to be increased by one.
 * Make sure that the old deprecated function works by calling the new function - don't have duplicated code!
 * The commit message should contain an `API` prefix (see ["commit message format"](code#commit-messages))
 * Document the change in the [changelog](/changelogs) for the next release
 * Deprecated APIs can be removed only after developers have had sufficient time to react to the changes. Hence, deprecated APIs should be removed in MAJOR releases only. Between MAJOR releases, leave the code in place with a deprecation warning. 
 * Exceptions to the deprecation cycle are APIs that have been moved into their own module, and continue to work with the new minor release. These changes can be performed in a single minor release without a deprecation period.
-* Add a `warnings` entry to the module's `.upgrade.yml` file so the
-[Silverstripe Upgrader can flag deprecated APIs](/upgrading/upgrading_module#upgradeyml).  
+
+When deprecating a method:
+
+* Add the following docblock `@deprecated 1.2.3 Use anotherMethod() instead`
+* Deprecation::notice('1.2.3', 'Use anotherMethod() instead'); to the top of the method
+
+When deprecating a class:
+
+* Add the following docblock `@deprecated 1.2.3 Use AnotherClass instead`
+* Add `Deprecation::notice('1.2.3', 'Use AnotherClass instead', SCOPE_CLASS);` to the top of `__construct()`
+
+When deprecating config:
+
+* Add the following docblock `@deprecated 1.2.3 Use different_config instead`
+
+If there is no immediate replacement for deprecated code that is being called, either because the replacement is not available until the next major version, or because there is not a plan for there to be a replacement, wrap the call to the deprecated code in `Deprecation::withNoReplacement()` e.g.
```

Comment:
```suggestion
If there is no immediate replacement for deprecated code that is being called, either because the replacement is not available until the next major version, or because there is not a plan for there to be a replacement, the message should be "Will be removed without equivalent functionality to replace it."

 ### Avoiding deprecated API
 
 Wherever possible, once some API has been deprecated, we should stop using it. This allows projects to emit deprecation notices and have an accurate list of code that they need to change.
 
 In some cases, we may not yet have a replacement for code that is required for the current major - but that we still have to call and support internally. In these cases, since we can't stop calling the code, wrap the call to the deprecated code in `Deprecation::withNoReplacement()` e.g:
   wrap the call to the deprecated code in `Deprecation::withNoReplacement()` e.g.
```

Reply:
Updated